jobsdb/prebackup/prebackup_test.go

```diff
@@ -0,0 +1,194 @@
+
+func Test_DropSourceIdsHandler(t *testing.T) {
+	handler := DropSourceIds(func() []string { return []string{"one", "two"} })
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	txn, err := postgresql.DB.Begin()
+	if err != nil {
+		t.Fatalf("Failed to create transaction: %s", err)
+	}
+	const (
+		jobsTable      = "jobs"
+		jobStatusTable = "job_status"
+	)
+
+	// Given I have jobs and job status tables with 3 events
+	createTables(t, txn, jobsTable, jobStatusTable)
+	addJob(t, txn, jobsTable, jobStatusTable, "one", "succeeded")
+	addJob(t, txn, jobsTable, jobStatusTable, "two", "failed")
+	addJob(t, txn, jobsTable, jobStatusTable, "three", "aborted")
+
+	if rows := countRows(t, txn, jobsTable); rows != 3 {
+		t.Fatalf("Expected 3 rows in jobs table, got %d", rows)
+	}
+	if rows := countRows(t, txn, jobStatusTable); rows != 3 {
+		t.Fatalf("Expected 3 rows in job status table, got %d", rows)
+	}
+
+	// when I execute the dropSourceIds handler for the 2 source_ids
+	if err = handler.Handle(ctx, txn, jobsTable, jobStatusTable); err != nil {
+		t.Fatalf("Failed to handle: %s", err)
+	}
+
+	// then I end up with one event on each table
+	if rows := countRows(t, txn, jobsTable); rows != 1 {
+		t.Fatalf("Expected 1 row in jobs table, got %d", rows)
+	}
+	if rows := countRows(t, txn, jobStatusTable); rows != 1 {
+		t.Fatalf("Expected 1 row in job status table, got %d", rows)
+	}
+
+}
+func Test_DropSourceIdsHandler_Without_SourceIds(t *testing.T) {
+	handler := DropSourceIds(func() []string { return []string{} })
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	var txn *sql.Tx
+
+	// when I execute the dropSourceIds handler for an empty source ids list, a canceled context and nil transaction
+	err := handler.Handle(ctx, txn, "jobs", "job_status")
+
+	// then handler doesn't do anything and no error is returned
+	if err != nil {
+		t.Fatalf("Failed to handle: %s", err)
+	}
+
+}
+
+func TestMain(m *testing.M) {
+	code := withBlockOnHold(func() int {
+		return run(m)
+	})
+	os.Exit(code)
+}
+
+func run(m *testing.M) int {
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Printf("Could not connect to docker: %s", err)
+		return -1
+	}
+	cleanup := &testhelper.Cleanup{}
+	defer cleanup.Run()
+	postgresql, err = destination.SetupPostgres(pool, cleanup)
+	if err != nil {
+		log.Printf("Could not start postgres: %s", err)
+		return -1
+	}
+
+	fmt.Println("DB_DSN:", postgresql.DB_DSN)
+	return m.Run()
+}
+
+func withBlockOnHold(f func() int) int {
+	var hold bool
+	flag.BoolVar(&hold, "hold", false, "hold environment clean-up after test execution until Ctrl+C is provided")
+	flag.Parse()
+
+	res := f()
+
+	if !hold {
+		return res
+	}
+	fmt.Println("Test on hold, before cleanup")
+	fmt.Println("Press Ctrl+C to exit")
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	<-c
+	return res
+}
+
+func createTables(t *testing.T, txn *sql.Tx, jobsTable, jobStatusTable string) {
```

Comment:
Do we have migrations that would create these tables in production for us? If yes could we potentially use those in the test instead?

Reply:
I cannot tell. Why do we need production (redacted?) data for this though? It is a simple, straightforward, unit test.

Reply:
Hold on, I thought you were bringing up a real DB so I thought it was an integration test? If that's the case I think it makes sense to be as close to production as possible. If you can (and want) it'd be great to have the same schema as it is applied in production here to avoid human mistakes and possible drifts (as in one change applied in production but then we forgot to apply it here in the test schema). Not blocking comment btw :+1: